Backend/app/kafka-consumers/src/Consumer/Flow.hs

```diff
@@ -105,6 +110,66 @@ availabilityConsumer flowRt appEnv kafkaConsumer =
         start = pure ([], Nothing)
         extract = pure
 
+becknRequestConsumer :: BRProcessor.BecknRequestType -> L.FlowRuntime -> AppEnv -> Consumer.KafkaConsumer -> IO ()
+becknRequestConsumer becknRequestType flowRt appEnv kafkaConsumer = do
+  flip finally (Consumer.closeConsumer kafkaConsumer) $ do
+    readBecknRequestMessages kafkaConsumer
+      >>= riderDrainerWithFlow
+      >>= commitAllTopics
+  where
+    riderDrainerWithFlow messages = do
+      runFlowR flowRt appEnv $
+        generateGUID
+          >>= flip withLogTag (BRProcessor.becknRequestProcessor becknRequestType messages)
+      pure messages
+
+    commitAllTopics messages = case messages of
+      [] -> print ("Nothing to commit" :: Text)
+      _ -> do
+        mbErr <- Consumer.commitAllOffsets Consumer.OffsetCommit kafkaConsumer
+        whenJust mbErr $ \err -> print ("Error while commit: message: " <> show err :: Text)
+
+readBecknRequestMessages ::
+  Consumer.KafkaConsumer ->
+  IO [BR.BecknRequestKafka]
+readBecknRequestMessages kafkaConsumer = do
+  pollStartTime <- getCurrentTime
+  let currentTopic = BR.countTopicNumber pollStartTime
+  eitherRecords <- pollMessageLoop (currentTopic, 0, [])
+  let records = mapMaybe hush eitherRecords
+  mbDecodedRecords <- forM records $ \record -> do
+    case Consumer.crValue record of
+      Just value -> do
+        case A.eitherDecode @BR.BecknRequestKafka . LBS.fromStrict $ value of
+          Right v -> pure (Just v)
+          Left err -> throwIO (InternalError $ "Could not decode record: " <> show value <> "; message: " <> show err) -- only when beckn_request tabular type was changed
+      Nothing -> pure Nothing
+  pure $ catMaybes mbDecodedRecords
+  where
+    pollMessageLoop :: (Int, Int, [Either Consumer.KafkaError ConsumerRecordD]) -> IO [Either Consumer.KafkaError ConsumerRecordD]
+    pollMessageLoop (currentTopic, emptyMessagesCount, messages) = do
+      eMessage <- Consumer.pollMessage kafkaConsumer (Consumer.Timeout 500)
+      (continueLoop, updEmptyMessagesCount) <- predicate currentTopic emptyMessagesCount eMessage
+      let updMessages = eMessage : messages
+      if continueLoop then pollMessageLoop (currentTopic, updEmptyMessagesCount, updMessages) else pure updMessages
+
+    predicate currentTopic emptyMessagesCount message = do
+      now <- getCurrentTime
+      if BR.countTopicNumber now /= currentTopic
+        then do
+          print ("Close consumer: current partition changed, not all messages was read!" :: Text)
```

Comment:
as we are planning to run it as a cronJob, lets let it run even if the hour window is passed (ideally this case should never come, but whats the harm in keeping it running if the records are still there ?).

Reply:
This case wasn't handled in logic. For example if we run Cron job at 10:30, then we subscribe on topics 0,1,2 ..8, 9,11, 12,23 (all except 10). When time cross to the next hour, we should change subscription, or create new consumer. Because if we will not change subscription, then consumer will fetch current messages and never stop. This is not handled in my PR

Reply:
@roman-bodavskiy discussed this with Piyush. We can keep this as is - it should work we think.

Also, while we are making changes here, can we see how to make this generic, to add more tables to push to S3?